src/lib/middleware/jwt-unused/jwt-unused.middleware.ts

```diff
@@ -0,0 +1,53 @@
+import express from "express";
+import { getLastNonce, updateJWTNonce } from "../../utils/database/DB";
+import { ExpressRequestWithToken } from "../../utils/auth/auth0";
+import { CONFIG } from "../../server/config/config.utils";
+import { createOrPropagateError } from "../../server/errors/errors.utils";
+import { OthentErrorID } from "../../server/errors/error";
+
+// TODO: This logic could be added to the `isRevoked` function / property of `express-jwt`:
+
+export function jwtUnusedFactory() {
+  return async (
+    req: ExpressRequestWithToken,
+    res: express.Response,
+    next: express.NextFunction,
+  ) => {
+    if (!CONFIG.MONGODB_ENABLED) {
+      // Skip the nonce check when running locally:
+      return next();
+    }
+
+    const { idToken } = req;
+
+    if (!idToken || !idToken.iat || !idToken.sub) {
+      throw createOrPropagateError(
+        OthentErrorID.Validation,
+        400,
+        "Invalid token data",
+      );
+    }
+
+    const lastNonce = await getLastNonce(idToken.sub);
```

Comment:
There's a potential issue here if 2 operations are performed at the same time and they make it to the server out-of-order, as processing the last one first will invalidate the other one.